tests/lib/nested.sh

```diff
@@ -0,0 +1,210 @@
+        PARAM_SMP="-smp 1"
+    fi
+
+    # TODO: enable ms key booting for i.e. nightly edge jobs ?
+    OVMF_CODE=""
+    OVMF_VARS=""
+    if [ "${ENABLE_SECURE_BOOT:-false}" = "true" ]; then
+        OVMF_CODE=".secboot"
+    fi
+    if [ "${ENABLE_OVMF_SNAKEOIL:-false}" = "true" ]; then
+        OVMF_VARS=".snakeoil"
+    fi
+
+    mkdir -p "${WORK_DIR}/image/"
+    cp -f "/usr/share/OVMF/OVMF_VARS${OVMF_VARS}.fd" "${WORK_DIR}/image/OVMF_VARS${OVMF_VARS}.fd"
+    PARAM_BIOS="-drive file=/usr/share/OVMF/OVMF_CODE${OVMF_CODE}.fd,if=pflash,format=raw,unit=0,readonly -drive file=${WORK_DIR}/image/OVMF_VARS${OVMF_VARS}.fd,if=pflash,format=raw"
+    PARAM_MACHINE="-machine q35${ATTR_KVM} -global ICH9-LPC.disable_s3=1"
+
+    if [ "${ENABLE_TPM:-false}" = "true" ]; then
+        TPMSOCK_PATH="/var/snap/swtpm-mvo/current/swtpm-sock"
+        if [ "${SPREAD_BACKEND}" = "lxd-nested" ]; then
+            mkdir -p /tmp/qtpm
+            swtpm socket --tpmstate dir=/tmp/qtpm --ctrl type=unixio,path=/tmp/qtpm/sock --tpm2 -d -t
+            TPMSOCK_PATH="/tmp/qtpm/sock"
+        elif ! snap list swtpm-mvo > /dev/null; then
+            snap install swtpm-mvo --beta
+            retry=60
+            while ! test -S /var/snap/swtpm-mvo/current/swtpm-sock; do
+                retry=$(( retry - 1 ))
+                if [ $retry -le 0 ]; then
+                    echo "Timed out waiting for the swtpm socket. Aborting!"
+                    return 1
+                fi
+                sleep 1
+            done
+        fi
+        PARAM_TPM="-chardev socket,id=chrtpm,path=${TPMSOCK_PATH} -tpmdev emulator,id=tpm0,chardev=chrtpm -device tpm-tis,tpmdev=tpm0"
+    fi
+
+    PARAM_IMAGE="-drive file=${IMAGE_FILE},cache=none,format=raw,id=disk1,if=none -device virtio-blk-pci,drive=disk1,bootindex=1"
+
+    # Create the systemd unit for running the VM, the qemu parameter order is 
+    # important. We run all of the qemu logic through a script that systemd runs
+    # for an easier time escaping everything
+    SVC_NAME="nested-vm-$(systemd-escape "${SPREAD_JOB:-unknown}")"
+    SVC_SCRIPT="/tmp/nested-vm-$RANDOM.sh"
+    cat << EOF > "${SVC_SCRIPT}"
+#!/bin/sh -e
+qemu-system-x86_64 \
+    ${PARAM_SMP} \
+    ${PARAM_CPU} \
+    ${PARAM_MEM} \
+    ${PARAM_TRACE} \
+    ${PARAM_LOG} \
+    ${PARAM_MACHINE} \
+    ${PARAM_DISPLAY} \
+    ${PARAM_NETWORK} \
+    ${PARAM_BIOS} \
+    ${PARAM_TPM} \
+    ${PARAM_RANDOM} \
+    ${PARAM_IMAGE} \
+    ${PARAM_SERIAL} \
+    ${PARAM_MONITOR}
+EOF
+    
+    chmod +x "${SVC_SCRIPT}"
+
+    printf '[Unit]\nDescription=QEMU Nested VM for UC20 spread job %s\n[Service]\nType=simple\nExecStart=%s\n' \
+        "${SPREAD_JOB:-unknown}" \
+        "${SVC_SCRIPT}" \
+            > "/run/systemd/system/${SVC_NAME}.service"
+    
+    systemctl daemon-reload
+    systemctl start "${SVC_NAME}"
+
+    # wait for the nested-vm service to appear active
+    if ! wait_for_service "${SVC_NAME}"; then
```

Comment:
Any reason why not just using `systemctl start --wait "${SVC_NAME}"`?

Reply:
I believe there is a lot of code left-overs as most of this is adapted from the snapd repository, which could explain the wait_for_service being flexible, even though we do not use any other state than active in this script. I'll try to run it with your suggestions and see if it runs correctly